### ISSUE TITLE
update tests to handle asserts more cleanly

### ISSUE THREAD

**Issue by rjrodger** [senecajs/seneca#134]
seneca captures callback exceptions for logging...

[Trello Card](https://trello.com/c/VUegrUVW/101-seneca-update-tests-to-handle-asserts-more-cleanly)

**Comment by paolochiodi** [comment]
How do you want to handle this?

In my opinion if no specific error handler is set seneca shouldn't catch the error or should re-throw it, with the same behaviour of node process `uncaughtException` event - and with the same gotchas.

**Comment by rjrodger** [comment]
needs some thinking by @geek - key issue is we're trying to solve two conflicting goals:
a) capture any issues in a standalone microservice
b) behave like a normal node module in other contexts, such as unit testing

**Comment by paolochiodi** [comment]
I don't get why we want a).
Isn't an accepted best practice to just exit the process in case of exception?

**Comment by rjrodger** [comment]
the exception needs to be captured so you can report it
the microservice may be running in a very ephemeral way - you can't even assume there'll be a log file to look at afterwards

also, it's important for the system as a whole to know that there are errors - especially once you get to 100s of instances

**Comment by PhilWaldmann** [comment]
currently you can't test your seneca actions (!!!!!), because 
```js
Lab.test('example', (done) => {
  seneca.act('role:example,cmd:test', function(){
    Code.expect('something').to.be.null() //should fail!
    done();
  });
});
```
will always timeout. (if `role:example,cmd:test` exists)

**Comment by rjrodger** [comment]
@PhilWaldmann we introduced a convenience method to handle this, try:

```
Lab.test('example', (done) => {
  seneca.error(done)
  seneca.act('role:example,cmd:test', function(){
    Code.expect('something').to.be.null() //should fail!
    done();
  });
});
```

The assert will throw, Seneca will capture this, and then pass to the done callback.

**Comment by rjrodger** [comment]
Also, for debugging, try:
```
SENECA_LOGS=all npm test
```

**Comment by atoi** [comment]
Seems like `seneca.error` does not help when you expect an error to occur. The following test always fails:

```javascript
lab.test('Action "role: test, cmd: create" does not accept an empty title', done => {
  const title = '';

  seneca
    .error(done)
    .act('role: test, cmd: create', { entity: { title } }, error => {
      expect(error).to.exist();
      done();
    });
});
```

**Comment by rjrodger** [comment]
in this case, either remove seneca.error(done) , or use a custom capture function for seneca.error that ignores your expected error

**Comment by atoi** [comment]
Thanks, that is what I was going to do. Just wanted to ensure there's no other way to achieve the desired behavior.

**Comment by mcdonnelldean** [comment]
Related to #397 and #398

**Comment by mcdonnelldean** [comment]
Merged with #398. Pretty much the feature set we are trying to add for vnext.
